In [21]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import statistics
%matplotlib inline

In [22]:
df=pd.DataFrame()
df['BB_age']=[14, 12, 11, 10, 8, 6, 8]

In [23]:
#Calculate Mean & Median
mean = np.mean(df['BB_age'])
median = np.median(df['BB_age'])
print(mean)
print(median)

9.85714285714
10.0


In [24]:
#Calculate Mode
(values, counts) = np.unique(df['BB_age'], return_counts=True)
ind = np.argmax(counts)
print(ind)
values[ind]

1


8

In [25]:
#Calculate variance, Standard deviation & standard error
a= np.var(df['BB_age'])
b= np.std(df['BB_age'],ddof=1)
c= b/np.sqrt(len(df['BB_age']))
print(a)
print(b)
print(c)

6.40816326531
2.73426232761
1.03345401972


2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

Estimate of Central Tendency:
I would choose the median among mean, median & mode to etimate the central tendency as it is more representative of the sample. The mode is two low in this case and will not tell us the average of the age and the mean, although close to the median is altered due to the extreme low value (6).

Etimate of Variance:
To describe the variance I would use the standard deviation. I will pick up the standard deviation as the standard error shows the meanigfulness of the estimated mean and in this case my choice is to pick up the median as the  most representative central tendency estimateof the sample.

Note: Having said so, the difference between the mean and the median is very little. Moreover, age can only be an integer, hence it would be 10 in both cases to be meaningful. If the mean is used as the central tendency estimate then I would choose the standard error


In [26]:
df2=pd.DataFrame()
df2['BB_age']=[14, 12, 11, 10, 8, 7, 8]

In [27]:
#Calculate Mean & Median
mean = np.mean(df2['BB_age'])
median = np.median(df2['BB_age'])
print(mean)
print(median)

10.0
10.0


In [28]:
#Calculate Mode
(values, counts) = np.unique(df2['BB_age'], return_counts=True)
ind = np.argmax(counts)
print(ind)
values[ind]

1


8

In [29]:
#Calculate variance, Standard deviation & standard error
a= np.var(df2['BB_age'])
b= np.std(df2['BB_age'],ddof=1)
c= b/np.sqrt(len(df2['BB_age']))
print(a)
print(b)
print(c)

5.42857142857
2.51661147842
0.951189731211


3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

Estimates that have not changed:
    
Median & mode

Estimates that have changed:

Mean, variance, standard deviation & standard error (from the variance estimates, variance is the one that experiments a larger change due to its characteritics)


In [30]:
df3=pd.DataFrame()
df3['BB_age']=[14, 12, 11, 10, 8, 7, 1]

In [31]:
#Calculate Mean & Median
mean = np.mean(df3['BB_age'])
median = np.median(df3['BB_age'])
print(mean)
print(median)

9.0
10.0


In [32]:
#Calculate Mode
(values, counts) = np.unique(df3['BB_age'], return_counts=True)
ind = np.argmax(counts)
print(ind)
values[ind]

0


1

In [33]:
#Calculate variance, Standard deviation & standard error
a= np.var(df3['BB_age'])
b= np.std(df3['BB_age'],ddof=1)
c= b/np.sqrt(len(df3['BB_age']))
print(a)
print(b)
print(c)

15.4285714286
4.24264068712
1.60356745147


4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

I would mantain the central tendency method: median as it doesn´t change and we can see how the mean now deviates from the original value due to the lower value that we have introduced
I would keed the standard deviation as both the standard deviation and the standard error have experienced a 68% change.


In [34]:
df4=pd.DataFrame()
df4['Adult_population']=[0.20, 0.23, 0.17, 0.05]

In [35]:
#Calculate Mean & Median
mean = np.mean(df4['Adult_population'])
median = np.median(df4['Adult_population'])
print(mean)
print(median)

0.1625
0.185


In [36]:
#Calculate Mode
(values, counts) = np.unique(df4['Adult_population'], return_counts=True)
ind = np.argmax(counts)
print(ind)
values[ind]

0


0.050000000000000003

In [37]:
#Calculate variance, Standard deviation & standard error
a= np.var(df4['Adult_population'])
b= np.std(df4['Adult_population'],ddof=1)
c= b/np.sqrt(len(df4['Adult_population']))
print(a)
print(b)
print(c)

0.00466875
0.0788986691903
0.0394493345951


5. Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

I would say that the & of adult population that were fans of Brandy Bunch show was 18.5% (median). I think we cannot use the mean as we have 5% low compared to the rest of the values. Hence the mean would be distorted by this value.

Additionally, the SciPhi fans 5% are not representative of the American Adults population but of a subset if it. Hence they can be excluded from the sample and the central tendency estimations will be more accurate. (Case 2)

Case 2: Excluding SciPhi fans (5%)

In [38]:
df4=pd.DataFrame()
df4['Adult_population']=[0.20, 0.23, 0.17]

#Calculate Mean & Median
mean = np.mean(df4['Adult_population'])
median = np.median(df4['Adult_population'])
print(mean)
print(median)

0.2
0.2


In [39]:
#Calculate Mode
(values, counts) = np.unique(df4['Adult_population'], return_counts=True)
ind = np.argmax(counts)
print(ind)
values[ind]

0


0.17000000000000001

In [40]:
#Calculate variance, Standard deviation & standard error
a= np.var(df4['Adult_population'])
b= np.std(df4['Adult_population'],ddof=1)
c= b/np.sqrt(len(df4['Adult_population']))
print(a)
print(b)
print(c)

0.0006
0.03
0.0173205080757


In this case (without considering the SciPhi fans), mean and meadian are equal equating to 20%. Hence the % of American adult population that were fans of Brandy Bunch show would be 20%.